apps/aechannel/src/aesc_fsm.erl

```diff
@@ -2740,20 +2748,27 @@ verify_signatures(Pubkeys, SignedTx) ->
     case aetx_sign:verify_half_signed(Pubkeys, SignedTx) of
         ok ->
             ok;
-        _ -> {error, wrong_signature}
+        _ -> {error, bad_signature}
     end.
 
-check_type_and_verify_signatures(SignedTx, Type, Pubkeys, ErrTypeMsg) ->
-    case aetx:specialize_type(aetx_sign:tx(SignedTx)) of
-        {Type, _Tx} ->
-            verify_signatures(Pubkeys, SignedTx);
+check_tx_and_verify_signatures(SignedTx, Type, ChannelPubkey, Pubkeys, ErrTypeMsg) ->
+    Aetx = aetx_sign:tx(SignedTx),
+    case aetx:specialize_type(Aetx) of
+        {Type, Tx} -> % check type
+            {Mod, Tx} = aetx:specialize_callback(Aetx),
+            case Mod:channel_pubkey(Tx) of
+                ChannelPubkey -> % expected pubkey
+                    verify_signatures(Pubkeys, SignedTx);
+                _ ->
```

Comment:
I see all `:channel_pubkey`s return pubkey without any tuple wrapper so correct. For readability, I would have preferred here a guard on the discarded value being a binary though it may be overkill. Ok.

I also see there is a test checking error `different_channel_id`, that makes reviewer more confident. Though I see many txs that expose `channel_pubkey` function, but it is not immediate to see how many of those are meant to be tested, and how many are tested (I checked #2132 too). May I suggest, in the list of test cases in `channel_ids`, to add comment mentioning which txs are knowingly not tested. (I am concerned that pull requests making the state machine aware of more transactions forget to integrate the tests in this test group.)

Reply:
I see comment added in following commit.